test/e2e/grpc_test.go

```diff
@@ -0,0 +1,78 @@
+//go:build e2e
+
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package test
+
+import (
+	"bufio"
+	"io"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/test/app"
+	"github.com/stretchr/testify/require"
+)
+
+func waitUntilGrpcReady(t *testing.T, serverApp *exec.Cmd, outputPipe io.ReadCloser) func() string {
```

Comment:
This looks almost identical to waitUntilReady in http_test.go. Perhaps we could extract this into a shared app.go, like app.WaitUntilReady (or app.WaitReady)?